### DISCUSSION TITLE
Is there a way to cross reference the table of contents in a quarto book?

### DISCUSSION THREAD

**Discussion by yyin422** [quarto-dev/quarto-cli#10867] [Q&A] [answered]
### Description

I'm trying to build a "back to toc" button in my pdf quarto book. Does quarto have something similar to `[xxx](#toc)` in Rmd using bookdown? Thank you!

**Comment by mcanouil** [comment, marked as answer]
As far as I know, there isn't for LaTeX.
For HTML, `[xxx](#TOC)` should work (I have no idea why the id is capitalised here).
For Typst, there isn't.

Here is a demo/workaround:

````qmd
---
title: "Quarto Playground"
format:
  html: default
  pdf:
    include-before-body:
      - text: |
          \addtocontents{toc}{\protect\hypertarget{toc}{}}
format-links: false
toc: true
toc-location: body
---

# Section 1

This is a playground for Quarto.

{{< lipsum 1 >}}

![An image]({{< placeholder 600 400 >}}){#fig-placeholder}

# Section 2

{{< lipsum 2 >}}

::: {.content-visible when-format="html"}
[Table of Contents](#TOC)
:::

::: {.content-visible when-format="pdf"}
`\hyperlink{toc}{Table of Contents}`{=latex}
:::
````

Adding `<toc>` in the Typst block in the template could make this work.
- https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/formats/typst/pandoc/quarto/typst-template.typ#L80-L86

I think, this can be a good enhancement.
CC @cscheid @gordonwoodhull @cderv

**Reply by cderv** [reply]
Probably worth opening this has an issue to track for feature request to make `[Table of Contents](#TOC)` works in all format possible by adding an id on TOC element. It does not seem that difficult to support 🤔 

Though, it could also be a Pandoc feature request as it may be pandoc adding an id in the first place.

**Reply by mcanouil** [reply]
Typst/HTML are easy. The tricky one (as usual) is LaTeX.
In my example, I had to use raw LaTeX (although I did not spend too much time on it).

**Reply by yyin422** [reply]
Thank you for the reply! The workaround is very helpful!
